### PR TITLE
Initial version for oc-react-template

### PR DESCRIPTION
**Pull request by nickbalestra**
Note: CSS not yet supported

### CODE REVIEW

File: blueprint/src/package.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+  "name": "base-component",
+  "description": "oc-template-react base component",
+  "version": "0.0.1",
+  "repository": "https://github.com/opentable/oc-template-react",
```
Comment (matteofigus) [inline blueprint/src/package.json]: I think I would remove repository? Just to ensure people don't use this my mistake

File: blueprint/src/package.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{
+  "name": "base-component",
+  "description": "oc-template-react base component",
+  "version": "0.0.1",
```
Comment (matteofigus) [inline blueprint/src/package.json]: What about 1.0.0?

File: blueprint/src/package.json (hunk 1 above)
Comment (matteofigus) [inline blueprint/src/package.json]: I think I would remove repository? Just to ensure people don't use this by mistake

File: blueprint/src/package.json (hunk 2 above)
Comment (nickbalestra) [inline blueprint/src/package.json]: npm correctly threat every version that is below 1.0.0, but we could go with 1.0.0 agree

File: blueprint/src/package.json [hunk 3]
```diff
@@ -18,10 +17,9 @@
         "example": "World",
         "default": "React"
       }
-    },
-    "state": "experimental"
+    }
   },
   "dependencies": {
-    "oc-template-react": "^0.0.2"
+    "oc-template-react": "^0.0.4"
```
Comment (matteofigus) [inline blueprint/src/package.json]: I think this should always be the same as version, innit?

File: package.json [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    "babel-preset-babili": "0.0.11",
+    "babel-preset-es2015": "^6.22.0",
+    "babel-preset-react": "^6.23.0",
+    "babili-webpack-plugin": "0.0.10",
+    "fs-extra": "^2.0.0",
+    "memory-fs": "^0.4.1",
+    "react": "^15.4.2",
+    "react-dom": "^15.4.2",
+    "uuid": "^3.0.1",
+    "webpack": "^2.2.1"
```
Comment (matteofigus) [inline package.json]: all of these should be dev dependencies right? If this is gonna be included on server-side node clients, these modules should kind of be dependency free?

Comment (matteofigus) [inline package.json]: I mean. I think the compile stuff should be all a 'devDependency' thing and when you do oc init this should do a full npm i (which will include dev dependencies). So the only `dependencies` stuff should be the `render` or `getCompiledTemplate` needed stuff (which in fact should kind of turn out to be dep free or so)

Comment (nickbalestra) [inline package.json]: Apart few differences in what get installed when, should be safe to say that `dependencies` are required to run, `devDependencies` only to develop it. In this direction for example, most of those are required dependencies for the `compile` method of the template, therefore should be dependencies

**Comment by nickbalestra** [comment]
@matteofigus all the dependencies now follow this convention:

- compileDependencies: modules needed to compile an oc-component based on the template, to be used with the cli on `oc dev`
- devDependencies: template level development/testing modules

This goes with https://github.com/opentable/oc/pull/480

File: src/getInfo.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+/* eslint-disable dot-notation */
+
+const packageJson = require('../package.json');
+
+module.exports = () => ({
+  type: packageJson.name,
+  version: packageJson.version,
+  dependencies: packageJson.compileDependencies,
```
Comment (matteofigus) [inline src/getInfo.js]: as discussed, I think removing the dependencies field is safe and useful as it will facilitate next step

**Pull request merged**